File: python/hamming/hamming.py

def distance(strand_a, strand_b):
    a = [letter for letter in strand_a]
    b = [letter for letter in strand_b]
    if len(strand_a) != len(strand_b):
        raise ValueError('lengths of both given strands are not the equal')

    else:
        hamming = []
        for n in range(len(a)):
            if a[n] != b[n]:
                hamming.append(a[n])
        return len(hamming)

File: go/hamming/hamming.go
// Package hamming calculate the hamming distance between two DNA strands
package hamming

import "errors"

// Distance calculate the hamming distance
func Distance(a, b string) (int, error) {

	ar, br := []rune(a), []rune(b)

	if len(ar) != len(br) {
		return 0, errors.New("lengths are not equal")
	}

	count := 0
	for n := range ar {
		if ar[n] != br[n] {
			count++
		}
	}
	return count, nil
}
